src/main/java/fi/aalto/cs/apluscourses/intellij/model/ExercisesUpdater.java

```diff
@@ -0,0 +1,63 @@
+package fi.aalto.cs.apluscourses.intellij.model;
+
+import fi.aalto.cs.apluscourses.intellij.notifications.DefaultNotifier;
+import fi.aalto.cs.apluscourses.intellij.notifications.NetworkErrorNotification;
+import fi.aalto.cs.apluscourses.intellij.notifications.Notifier;
+import fi.aalto.cs.apluscourses.intellij.services.PluginSettings;
+import fi.aalto.cs.apluscourses.utils.Event;
+import fi.aalto.cs.apluscourses.utils.async.RepeatedTask;
+import java.io.IOException;
+import org.jetbrains.annotations.NotNull;
+
+public class ExercisesUpdater extends RepeatedTask {
+
+  private final CourseProject courseProject;
+
+  private final Event eventToTrigger;
+
+  private final Notifier notifier;
+
+  /**
+   * Construct an updater with the given parameters.
+   */
+  public ExercisesUpdater(@NotNull CourseProject courseProject,
+                          @NotNull Event eventToTrigger,
+                          @NotNull Notifier notifier,
+                          long updateInterval) {
+    super(updateInterval);
+    this.courseProject = courseProject;
+    this.eventToTrigger = eventToTrigger;
+    this.notifier = notifier;
+  }
+
+  public ExercisesUpdater(@NotNull CourseProject courseProject,
+                          @NotNull Event eventToTrigger) {
+    this(courseProject, eventToTrigger, new DefaultNotifier(), PluginSettings.UPDATE_INTERVAL);
+  }
+
+  @Override
+  protected void doTask() {
+    var course = courseProject.getCourse();
+    var dataSource = course.getExerciseDataSource();
+    var authentication = courseProject.getAuthentication();
+    if (authentication == null) {
+      return;
+    }
+    try {
+      var points = dataSource.getPoints(course, authentication);
+      if (Thread.interrupted()) {
+        return;
+      }
+      points.setSubmittableExercises(course.getExerciseModules().keySet()); // TODO: remove
```

Comment:
Should this be removed if it's marked as "TODO: remove"? ;p

Reply:
Yes eventually, but this is copied from `MainViewModel`, so the comment is still relevant but it's a bigger fix :smile: